airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -208,18 +213,28 @@ private CancellationScope generateSyncWorkflowRunnable(final ConnectionUpdaterIn
       workflowInternalState.setJobId(getOrCreateJobId(connectionUpdaterInput));
       workflowInternalState.setAttemptNumber(createAttempt(workflowInternalState.getJobId()));
 
-      final GeneratedJobInput jobInputs = getJobInput();
+      final int generateCheckInputVersion =
+          Workflow.getVersion(GENERATE_CHECK_INPUT_TAG, Workflow.DEFAULT_VERSION, GENERATE_CHECK_INPUT_CURRENT_VERSION);
+
+      GeneratedJobInput jobInputs = null;
+      if (generateCheckInputVersion < GENERATE_CHECK_INPUT_CURRENT_VERSION) {
+        jobInputs = getJobInput();
+      }
 
       reportJobStarting(connectionUpdaterInput.getConnectionId());
       StandardSyncOutput standardSyncOutput = null;
 
       try {
-        final SyncCheckConnectionFailure syncCheckConnectionFailure = checkConnections(jobInputs);
+        final SyncCheckConnectionFailure syncCheckConnectionFailure = checkConnections(getJobRunConfig(), jobInputs);
         if (syncCheckConnectionFailure.isFailed()) {
           final StandardSyncOutput checkFailureOutput = syncCheckConnectionFailure.buildFailureOutput();
           workflowState.setFailed(getFailStatus(checkFailureOutput));
           reportFailure(connectionUpdaterInput, checkFailureOutput, FailureCause.CONNECTION);
         } else {
+          if (generateCheckInputVersion >= GENERATE_CHECK_INPUT_CURRENT_VERSION) {
+            jobInputs = getJobInput();
+          }
```

Comment:
This is where we are moving the activity that generates the SyncWorkflow inputs to after the potential checks. Based on the workflow version, we keep the old position of the activity.

We do this so that any config updates that happen during check are freshly retrieved before running the sync.